### PR TITLE
Niksseif intro to destructuring test

### PR DESCRIPTION
**Pull request by niksseif**
Image paths are added.

### CODE REVIEW

File: lessons/module-2/intro-to-destructuring.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+console.log(color1)
+console.log(color2)
+```
+ 
+
+<section class="answer">
+### The Answer <br>
+This is what happen in the code above. 
+<img class="medium" src="https://github.com/turingschool/front-end-curriculum/blob/f3b5d757b797c33fc08ff93c80c9122f875f2a3a/assets/images/lessons/intro-to-destructuring/intro-to-destructuring-array.png" />
```
Comment (Kalikoze) [inline lessons/module-2/intro-to-destructuring.md]: This is super close, I think the file path will work if it's a relative file path.  Double check the naming of the folder (it's intro-todestructuring without the extra -) and then make sure that the image name has no spaces.  

So an example could be ``

File: lessons/module-2/intro-to-destructuring.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+```
+</section>
+
+<section class="answer">
+    
+### The Answer   
+    
+Assigning an Array to a variable
+    
+<img class="medium" src="https://github.com/turingschool/front-end-curriculum/blob/dfec040a99f1673538180127f053d7753b3c9a8e/assets/images/lessons/intro-to-destructuring/intro-to-destructuring-array-example1.png"/>
```
Comment (Kalikoze) [inline lessons/module-2/intro-to-destructuring.md:67]: Same scenario here, we can use the relative file path and then make sure you are using dashes in the image name.

`   `

File: lessons/module-2/intro-to-destructuring.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+colors = ["Red","Pink","Blue","Black"] //Array to be destructured
+
+var [color1, color2] = colors //Destructuring
+
+console.log(color1)
+console.log(color2)
+```
+ 
+
+<section class="answer">
```
Comment (Kalikoze) [inline lessons/module-2/intro-to-destructuring.md:28]: Love that you're using these sections to hide/display answers!

File: lessons/module-2/intro-to-destructuring.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    };
+
+// Destructuring the object into variables without 
+// assigning default values 
+var { firstname, lastname, country } = user;
+console.log("Without setting default values")
+console.log( firstname, lastname, country);
+
+// Destructuring the object into variables by 
+// assigning default values
```
Comment (Kalikoze) [inline lessons/module-2/intro-to-destructuring.md:150]: I like how you continue to build on the previous example and add more options that can be done.  Very cool!

File: lessons/module-2/intro-to-destructuring.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+// Destructuring the object into variables by 
+// assigning default values 
+var { firstname = 'default firstname', 
+      lastname = 'default lastname', 
+      country = 'default country' } = employee;
+console.log("\n After setting default values")
+console.log( firstname, lastname, country);
+```
+#### Passing an object as an argument:
+What if you want to destrcuture an object as an argument?
```
Comment (Kalikoze) [inline lessons/module-2/intro-to-destructuring.md]: Slight typo with "destructure".  But thank you for adding in an example for destructuring an argument in a function!

**Review by Kalikoze — CHANGES_REQUESTED** [review]
Overall this looks great Nik!  As mentioned in my notes, I would just double check the file path so that it matches the exact folder name "intro-todestructuring" and remove any spaces in the images.  I really appreciate how examples build off each other and how you're having students lead the conversations and learning!

File: lessons/module-2/intro-to-destructuring.md (hunk 1 above)
Comment (niksseif) [reply lessons/module-2/intro-to-destructuring.md]: Hmm... I am using Github copy path button to copy the path of the image! It is not working, but when I use permalink it works:)
[https://github.com/turingschool/front-end-curriculum/blob/gh-pages/assets/images/lessons/intro-to-destructuring/intro-to-destructuring-array.png](url)

File: lessons/module-2/intro-to-destructuring.md (hunk 2 above)
Comment (niksseif) [reply lessons/module-2/intro-to-destructuring.md:67]: Hmmmm... I somehow made it work with permalink. I am not sure where you find that intro-todestructuring folder? I am trying to access the images from this folder intro-to-destructuring. I'll trouble shooting this. Thank you

**Pull request merged**